### PR TITLE
Create admin auths API

### PR DESCRIPTION
**Pull request by daniil-pankratov**
This PR adds new admin auths API endpoints: one - for getting list of existing authentication sources with their configuration and another one - to create new authentication source.

### CODE REVIEW

File: routers/api/v1/admin/auths.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package admin
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
```
Comment (6543) [inline routers/api/v1/admin/auths.go:11]: sort import order

Comment (6543) [reply routers/api/v1/admin/auths.go:11]: -> https://github.com/go-gitea/gitea/blob/master/CONTRIBUTING.md#styleguide

Comment (daniil-pankratov) [reply routers/api/v1/admin/auths.go:11]: Done

**Comment by lunny** [comment]
Some config may contains secrets which should not be expose?

**Comment by yardenshoham** [comment]
Is this PR stale?

**Comment by wxiaoguang** [comment]
I guess there could be some obstacles for this PR at the moment  (just a guess):

1. security concern (might be resolved by scoped token / privileged token)
2. API stability: Gitea's auth system has too many technical debts, exposing these auth sources directly seems not be able to be stable. I can foresee that Gitea's auth system needs to be refactored in near future to support: hide local login/set default login/2FA policy, etc.

**Pull request closed without merging**